do-not-commit/ssrf-requests.py

```diff
@@ -0,0 +1,39 @@
+import flask
+import requests
+
+app = flask.Flask(__name__)
+
+
+@app.route("/route_param/<route_param>")
+def route_param(route_param):
+    print("blah")
+    # ruleid: ssrf-requests
+    return requests.get(route_param)
+
+
+@app.route("/get_param", methods=["GET"])
+def get_param():
+    param = flask.request.args.get("param")
+    # ruleid: ssrf-requests
+    requests.post(param, timeout=10)
+
+
+@app.route("/post_param", methods=["POST"])
+def post_param():
+    param = flask.request.form["param"]
+    if True:
+        # ruleid: ssrf-requests
+        requests.get(param)
+
+
+@app.route("/subexpression", methods=["POST"])
+def subexpression():
+    param = "{}".format(flask.request.form["param"])
+    print("do things")
+    # ruleid: ssrf-requests
+    requests.post(param, data={"hello", "world"})
```

Comment:
By default, 'requests' calls wait until the connection is closed.
This means a 'requests' call without a timeout will hang the program
if a response is never received. Consider setting a timeout for all
'requests'. ([actice.use-timeout.use-timeout](https://semgrep.dev/r/actice.use-timeout.use-timeout))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=actice.use-timeout.use-timeout) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>